qiskit/circuit/annotated_operation.py

```diff
@@ -0,0 +1,169 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Annotated Operations."""
+
+import dataclasses
+from typing import Union, List
+
+from qiskit.circuit.operation import Operation
+from qiskit.circuit._utils import _compute_control_matrix, _ctrl_state_to_int
+from qiskit.circuit.exceptions import CircuitError
+
+
+class Modifier:
+    """Modifier class."""
+
+    pass
+
+
+@dataclasses.dataclass
```

Comment:
The `__repr()__` function is a bit different: for instance `print(InverseModifier())` outputs `<qiskit.circuit.annotated_operation.InverseModifier object at 0x00000233A87C0160>` without the decorator, and `InverseModifier()` with the decorator. This is not a big deal, but it would probably make sense to keep this consistent across different modifiers.